contracts/starknet/teleport_join.cairo

```diff
@@ -0,0 +1,938 @@
+%lang starknet
+# pragma solidity 0.8.14;
+
+# import "./TeleportGUID.sol";
+from contracts.starknet.teleport_GUID import TeleportGUID, getGUIDHash
+from starkware.cairo.common.cairo_builtins import HashBuiltin, BitwiseBuiltin
+from starkware.starknet.common.syscalls import (
+    get_caller_address,
+    get_contract_address,
+    get_block_timestamp,
+)
+from starkware.cairo.common.uint256 import (
+    Uint256,
+    uint256_check,
+    uint256_le,
+    uint256_neg,
+    uint256_lt,
+)
+from contracts.starknet.safe_math import (
+    Int256,
+    add,
+    _add,
+    sub,
+    _sub,
+    mul,
+    _mul,
+    add_signed,
+    _min,
+    _felt_to_uint,
+    _uint_to_felt,
+)
+from contracts.starknet.assertions import eq_0, either, le_int, _ge_0, assert_either, is_eq, check
+
+struct Urn:
+    member ink : Uint256  # Locked Collateral  [wad]
+    member art : Uint256  # Normalised Debt    [wad]
+end
+
+# struct TeleportStatus {
+#         bool    blessed;
+#         uint248 pending;
+#     }
+struct TeleportStatus:
+    member blessed : felt
+    member pending : Uint256
+end
+
+# interface VatLike {
+#     function dai(address) external view returns (uint256);
+#     function live() external view returns (uint256);
+#     function urns(bytes32, address) external view returns (uint256, uint256);
+#     function frob(bytes32, address, address, address, int256, int256) external;
+#     function hope(address) external;
+#     function move(address, address, uint256) external;
+#     function nope(address) external;
+#     function slip(bytes32, address, int256) external;
+# }
+@contract_interface
+namespace VatLike:
```

Comment:
Order of methods should be the same as in the solidity comments above. @EvolveArt